### PR TITLE
Allow usage without Rails

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: lib/rescue_registry/exception_handler.rb [hunk 1]
```diff
@@ -10,6 +12,7 @@ def initialize(exception, **options)
 
       status = options[:status]
       if status == :passthrough
+        # FIXME: Won't work without Rails
         status = ActionDispatch::ExceptionWrapper.status_code_for_exception_without_rescue_registry(exception.class.name)
```
Comment (wagenet) [inline lib/rescue_registry/exception_handler.rb]: Still need to figure out how to handle this. Not sure it makes sense to have this option outside of Rails. If that's the case then we should make an API for this that only applies in Rails.

File: spec/rack_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  let :context do
+    Module.new do
+      include RescueRegistry::Context
+
+      register_exception CustomError, status: 403
+    end
+  end
+
+  before do
+    RescueRegistry.context = context
```
Comment (wagenet) [inline spec/rack_spec.rb]: Might be nice to have some sort of default context.

Comment (wagenet) [inline spec/rack_spec.rb]: We should probably cleanup the context when the spec is done.

File: lib/rescue_registry/show_exceptions.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    # Unfortunately, Rail's nice mime types are in ActionDispatch which we'd rather not require
+    CONTENT_TYPES = {
+      jsonapi: "application/vnd.api+json",
+      json:    "application/json",
+      xml:     ["application/xml", "text/xml"],
+      plain:   "text/plain"
+    }
+
+    # Match Rail's Mime::Type API on a basic level
+    MimeType = Struct.new(:to_sym, :to_s)
```
Comment (wagenet) [inline lib/rescue_registry/show_exceptions.rb]: This feels a little iffy, but I don't have a better way to do this without depending upon Action Dispatch.

Comment (richmolj) [reply lib/rescue_registry/show_exceptions.rb]: Works for me, having this class at all is 💯

File: spec/rack_spec.rb (hunk 2 above)
Comment (richmolj) [reply spec/rack_spec.rb]: maybe `around do / RescueRegistry.with_context`

File: lib/rescue_registry/exception_handler.rb (hunk 1 above)
Comment (wagenet) [reply lib/rescue_registry/exception_handler.rb]: For now I've abstracted it a bit based on whether `ActionDispatch::ExceptionWrapper` is defined. I think there's probably further room for improvement here.

File: lib/rescue_registry/context.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+module RescueRegistry
+  module Context
+    extend ActiveSupport::Concern
+
+    included do
+      cattr_accessor :rescue_registry
```
[the remark this answers is not in the document]
Comment (wagenet) [reply lib/rescue_registry/context.rb]: Good call. I think `cattr_acessor` is incorrect. Will add a test.

**Pull request merged**